### PR TITLE
Feat: Intent warning for events and prefixed commands

### PR DESCRIPTION
**Pull request by Kigstn**
## What type of pull request is this?

- [x] Non-breaking code change
- [ ] Breaking code change
- [ ] Documentation change/addition
- [ ] Tests change

## Description

This warns users if they use a gateway event / prefixed commands and do not have the required intents enabled. If users read their console output, this should make debugging easier for them.

Lists of intents and events taken from https://discord.com/developers/docs/topics/gateway#list-of-intents

## Changes

- warn when event is missing required intent
- warn when prefixed commands are used but the message content intent is missing

## Checklist

- [x] I've formatted my code with [Black](https://black.readthedocs.io/en/stable/)
- [x] I've ensured my code works on `Python 3.10.x`
- [x] I've tested my code

### CODE REVIEW

**Comment by LordOfPolls** [comment]
I'd suggest using references to the event objects rather than strings. That way if we were to refactor they'd be caught by most modern IDEs

**Comment by Kigstn** [comment]
> I'd suggest using references to the event objects rather than strings. That way if we were to refactor they'd be caught by most modern IDEs

Sure, good call

**Comment by AstreaTSS** [comment]
Too lazy to do a proper review, but it should be noted that prefixed commands will work fine even without the message content intent *unless* they change the prefixes used to outside of their defaults (you can always mention the bot). Probably worth checking for that.

File: naff/client/client.py [hunk 1]
```diff
@@ -33,7 +33,55 @@
 
 import naff.api.events as events
 import naff.client.const as constants
-from naff.api.events import RawGatewayEvent, MessageCreate
+from naff.api.events import (
```
Comment (silasary) [inline naff/client/client.py]: We are already importing the events namespace on line 34, let's just use that, and refer to events as `events.Whatever`, the same way we do when dispatching them.

**Review by silasary — CHANGES_REQUESTED** [review]

**Review by silasary — APPROVED** [review]

File: naff/client/client.py [hunk 2]
```diff
@@ -33,9 +33,7 @@
 
 import naff.api.events as events
 import naff.client.const as constants
-from naff.api.events import RawGatewayEvent, MessageCreate
-from naff.api.events import processors
-from naff.api.events.internal import Component, BaseEvent
+from naff.api.events import MessageCreate, RawGatewayEvent, processors, Component, BaseEvent
```
Comment (silasary) [inline naff/client/client.py:36]: Bonus points if you refactor all existing references to these to also use `events.`, but I'm happy to merge without that

**Pull request merged**